src/components/nfts/NftGrid/index.tsx

```diff
@@ -1,57 +1,204 @@
+                    headCell.label
+                  )}
+                </TableCell>
               ))}
-            </Grid>
-          </Box>
-        )
-      })}
+            </TableRow>
+          </TableHead>
+
+          <TableBody>
+            {nfts.map((item) => (
+              <TableRow tabIndex={-1} key={`${item.address}-${item.id}`} onClick={() => onSelect(item)}>
+                {/* Collection name */}
+                <TableCell>
+                  <Box display="flex" alignItems="center" alignContent="center" gap={1}>
+                    <Box width={iconSize} height={iconSize} mr={1}>
+                      <ImageFallback
+                        src={item.logoUri}
+                        alt={`${item.tokenName} collection icon`}
+                        fallbackComponent={fallbackIcon}
+                        fallbackSrc=""
+                        style={iconStyle}
+                      />
+                    </Box>
+
+                    <Typography>{item.tokenName || item.tokenSymbol}</Typography>
+                  </Box>
+                </TableCell>
+
+                {/* Token ID */}
+                <TableCell>
+                  <Box display="flex" alignItems="center" alignContent="center" gap={1}>
+                    <Box width={iconSize} height={iconSize} mr={1}>
+                      {item.imageUri ? (
+                        <ImageFallback
+                          src={item.imageUri}
+                          alt={`${item.tokenName} NFT preview`}
+                          fallbackComponent={fallbackIcon}
+                          fallbackSrc=""
+                          style={iconStyle}
+                        />
+                      ) : null}
+                    </Box>
+
+                    <ExternalLink href={linkTemplates ? linkTemplates[0].getUrl(item) : ''} onClick={stopPropagation}>
+                      <Typography sx={item.name ? undefined : { wordBreak: 'break-all' }}>
+                        {item.name || `${item.tokenSymbol} #${item.id.slice(0, 20)}`}
+                      </Typography>
+                    </ExternalLink>
+                  </Box>
+                </TableCell>
+
+                {/* Links */}
+                <TableCell>
+                  <Box display="flex" alignItems="center" alignContent="center" gap={1.5}>
+                    {linkTemplates?.map(({ title, logo, getUrl }) => (
+                      <ExternalLink href={getUrl(item)} key={title} onClick={stopPropagation} noIcon>
+                        <img src={logo} width={24} height={24} alt={title} />
+                      </ExternalLink>
+                    ))}
+                  </Box>
+                </TableCell>
+
+                {/* Checkbox */}
+                <TableCell align="right">
+                  <Checkbox checked={selectedNfts.includes(item)} />
+                </TableCell>
+              </TableRow>
+            ))}
+
+            {/* Fill up the table up to min rows */}
```

Comment:
Why do we need these min rows?

Is it related to the InfiniteScrolling?

Reply:
Yes. When you filter, and there are fewer than 10 rows, we don't want the page to jump and trigger InfiniteScroll.